### PR TITLE
Add `releaseNotesURL` setting

### LINKED ISSUE #7125: Add `releaseNotesURL` setting?
Scala Steward has adopted a `info.releaseNotesUrl` property for the POM.

https://contributors.scala-lang.org/t/add-release-notes-urls-to-your-poms/6059

Can we add a setting `releaseNotesURL` to populate it?

This would be similar to the `apiURL` setting which populates `info.apiURL` in the POM.
https://github.com/sbt/sbt/blob/973cd63e0b6bfe0d06d364f314f391a08a77d38b/main/src/main/scala/sbt/Keys.scala#L345

### PR DESCRIPTION
**Pull request by armanbilge**
Closes https://github.com/sbt/sbt/issues/7125

Scala Steward has adopted a `info.releaseNotesUrl` property for the POM.

https://contributors.scala-lang.org/t/add-release-notes-urls-to-your-poms/6059

Can we add a setting `releaseNotesURL` to populate it?

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @armanbilge,

Thank you for your contribution! We really value the time you've taken to put this together.

We see that you have signed the Lightbend Contributors License Agreement before, however, the CLA has changed since you last signed it.
Please review the new CLA and sign it before we proceed with reviewing this pull request:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

File: main/src/main/scala/sbt/Defaults.scala [hunk 1]
```diff
@@ -200,7 +200,8 @@ object Defaults extends BuildCommon {
       autoCompilerPlugins :== true,
       scalaHome :== None,
       apiURL := None,
-      javaHome :== None,
+      releaseNotesURL := None
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/Defaults.scala]: Missing comma here.

**Comment by eed3si9n** [comment]
I wonder if it would be more straightforward to derive the release notes URL by something like:

```scala
ThisBuild / sourceRepo := Some(GitHubRepo("foo/bar"))
```

**Comment by lightbend-cla-validator** [comment]
Hi @armanbilge,

Thank you for your contribution! We really value the time you've taken to put this together.

We see that you have signed the Lightbend Contributors License Agreement before, however, the CLA has changed since you last signed it.
Please review the new CLA and sign it before we proceed with reviewing this pull request:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

**Comment by armanbilge** [comment]
> I wonder if it would be more straightforward to derive the release notes URL by something like:

Sure, `scmInfo` could be derived from that as well. But that seems like a new feature / PR. :)

I suppose you are happy to move forward with this then? Still a handful of things to do, possibly in some upstream repos.

File: main/src/main/scala/sbt/Defaults.scala [hunk 2]
```diff
@@ -3344,7 +3345,12 @@ object Classpaths {
         p1.extra(SbtPomExtraProperties.VERSION_SCHEME_KEY -> x)
       case _ => p1
     }
-    p2
+    val p3 = releaseNotesURL.value match {
+      case Some(u) =>
+        p1.extra("info.releaseNotesUrl" -> u.toExternalForm)
```
Comment (armanbilge) [inline main/src/main/scala/sbt/Defaults.scala]: I assume this constant should be added to `SbtPomExtraProperties`, where is that exactly?

Comment (eed3si9n) [reply main/src/main/scala/sbt/Defaults.scala]: https://github.com/sbt/librarymanagement/blob/develop/core/src/main/java/sbt/internal/librarymanagement/mavenint/SbtPomExtraProperties.java

**Comment by eed3si9n** [comment]
> I suppose you are happy to move forward with this then? Still a handful of things to do, possibly in some upstream repos.

Yea. I am +1 on enriching POM for this purpose.

File: main/src/main/scala/sbt/Defaults.scala (hunk 2 above)
Comment (armanbilge) [reply main/src/main/scala/sbt/Defaults.scala]: - https://github.com/sbt/librarymanagement/pull/410

Comment (mzuehlke) [inline main/src/main/scala/sbt/Defaults.scala]:
```suggestion
        p2.extra("info.releaseNotesUrl" -> u.toExternalForm)
```
I think this must reference `p2` here.

**Comment by eed3si9n** [comment]
https://github.com/sbt/librarymanagement/releases/tag/v1.9.0-M1 should be on its way to Maven Central.

**Comment by armanbilge** [comment]
I guess this is good to go? I looked at `apiURL` but didn't see any tests I could use as an example.

**Comment by eed3si9n** [comment]
I think there are related scripted tests that you might be able to modify or copy-and-modify:
- https://github.com/sbt/sbt/blob/1.9.x/sbt-app/src/sbt-test/dependency-management/make-pom/build.sbt
- https://github.com/sbt/sbt/blob/1.9.x/sbt-app/src/sbt-test/dependency-management/evicted-semver-spec/build.sbt

**Comment by armanbilge** [comment]
Thanks! While I was here I added a test for `apiURL` as well.

**Review by eed3si9n — APPROVED** [review]
Thanks!

**Pull request merged**